github/actions_artifacts_test.go

```diff
@@ -0,0 +1,243 @@
+	}
+	if artifacts != nil {
+		t.Errorf("Actions.ListWorkflowRunArtifacts return %+v, want nil", artifacts)
+	}
+}
+
+func TestActionsService_GetArtifact(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/actions/artifacts/1", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		fmt.Fprint(w, `{"id":1, "node_id": "xyz", "name": "a", "size_in_bytes": 5, "archive_download_url": "u", "expired": false, "created_at": "b", "expires_at": "c"}`)
+	})
+
+	artifact, _, err := client.Actions.GetArtifact(context.Background(), "o", "r", 1)
+	if err != nil {
+		t.Errorf("Actions.GetArtifact returned error: %v", err)
+	}
+
+	want := &Artifact{ID: Int64(1), NodeID: String("xyz"), Name: String("a"), SizeInBytes: Int64(5), ArchiveDownloadURL: String("u"), Expired: Bool(false), CreatedAt: String("b"), ExpiresAt: String("c")}
+	if !reflect.DeepEqual(artifact, want) {
+		t.Errorf("Actions.GetArtifact returned %+v, want %+v", artifact, want)
+	}
+}
+
+func TestActionsService_GetArtifact_invalidOwner(t *testing.T) {
+	client, _, _, teardown := setup()
+	defer teardown()
+
+	_, _, err := client.Actions.GetArtifact(context.Background(), "%", "r", 1)
+	testURLParseError(t, err)
+}
+
+func TestActionsService_GetArtifact_invalidRepo(t *testing.T) {
+	client, _, _, teardown := setup()
+	defer teardown()
+
+	_, _, err := client.Actions.GetArtifact(context.Background(), "%", "r", 1)
+	testURLParseError(t, err)
+}
+
+func TestActionsService_GetArtifact_notFound(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/actions/artifacts/1", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		w.WriteHeader(http.StatusNotFound)
+	})
+
+	artifact, resp, err := client.Actions.GetArtifact(context.Background(), "o", "r", 1)
+	if err == nil {
+		t.Errorf("Expected HTTP 404 response")
+	}
+	if got, want := resp.Response.StatusCode, http.StatusNotFound; got != want {
+		t.Errorf("Actions.GetArtifact return status %d, want %d", got, want)
+	}
+	if artifact != nil {
+		t.Errorf("Actions.GetArtifact return %+v, want nil", artifact)
+	}
+}
+
+func TestActionsSerivice_DownloadArtifact(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/actions/artifacts/1/zip", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		w.Header()["Location"] = []string{"https://location.com/artifact"}
+		w.WriteHeader(http.StatusNoContent)
+	})
+
+	_, err := client.Actions.DownloadArtifact(context.Background(), "o", "r", 1, "zip")
+	if err != nil {
+		t.Errorf("Actions.DownloadArtifact returned error: %v", err)
+	}
+}
+
+func TestActionsService_DownloadArtifact_invalidOwner(t *testing.T) {
+	client, _, _, teardown := setup()
+	defer teardown()
+
+	_, err := client.Actions.DownloadArtifact(context.Background(), "%", "r", 1, "zip")
```

Comment:
What is this actually testing?  I think it can be deleted.

Reply:
https://github.com/google/go-github/pull/1414#discussion_r376749484